pkg/pillar/hypervisor/kvm.go

```diff
@@ -748,7 +748,7 @@ func (ctx KvmContext) Setup(status types.DomainStatus, config types.DomainConfig
 		"-readconfig", file.Name(),
 		"-pidfile", kvmStateDir+domainName+"/pid")
 
-	spec, err := ctx.setupSpec(&status, &config, status.OCIConfigDir)
+	spec, err := ctx.setupSpec(&status, &config, aa, globalConfig, status.OCIConfigDir)
```

Comment:
How are these new parameters relevant for kvm/xen if this PR is about bare metal containers?

Reply:
These parameters are already available and passed to the Setup function (which calls setupSpec()), they are pointers so no initialization is required, what is the rationale to initialize new variables and discard the arguments already available in the caller function?

Reply:
> what is the rationale to initialize new variables and discard the arguments already available in the caller function

because the real values are not needed and it is better not to pass information to functions that they don't need.
I think that's exactly the same what you mean.

On top of it, as far as I understand, it allows insecure defaults: A user who uses kvm virtualization can now create a container with devices forwarded into the container that qemu runs in, isn't it? For the user it has not advantage, only disadvantages as f.e if an intruder can break out of qemu, then the container boundary is even easier to penetrate.
Even if it is now implemented in a way that they have full access to all devices anyways, it may be forgotten once this is solved.

Reply:
done.